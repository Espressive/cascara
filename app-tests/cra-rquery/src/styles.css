html,
body,
#login {
  margin: 0.5em;
  padding: 0;
}

main {
  display: grid;
  justify-content: center;
  align-content: center;
  background-color: ghostwhite;
}

aside {
  grid-column: 1;
}

aside nav {
  display: flex;
  flex-direction: column;
}

aside nav a {
  line-height: 1.5em;
}

section {
  grid-column: 2;
}
