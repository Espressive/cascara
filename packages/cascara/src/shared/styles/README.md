# Cascara SASS

This will likely be only a collection of consistent styling patterns and mixins we want to use for the library. We are using SASS so we do not end up with vendor lock-in for our styles. There are dozens of options for CSS in JS out there. Some make it, and some do not. The risk is too great for us in trying to support one of these technologies for a platform like our design system.

We are using the latest version of Dart Sass. [Please review all documentation(https://sass-lang.com/documentation)] as there are some minor differences with LibSass and Ruby Sass which are older.

At some point, maybe it makes sense to move all tokens and build tooling for tokens here and make into a package to more easily include in FDS components. That time may come soon as we build more and see the limitations.
