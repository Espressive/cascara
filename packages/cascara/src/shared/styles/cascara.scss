@forward 'lib/scroll' as scroll-*;
@forward 'color/brand';

$test-var: var(--test);
