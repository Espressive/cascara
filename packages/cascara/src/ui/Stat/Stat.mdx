---
title: Stat
propTable: Stat.js
---

```jsx
<Stat label='Stat' sub='Component' />
```
