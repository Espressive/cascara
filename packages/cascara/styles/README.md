# Cascara SASS

Collection of variables and mixins for consistent styling.

## Getting Started

### Usage

To access the variables and mixins in this package, you must declare the library in your SCSS file.

```
@use '@espressive/cascara/styles' as *;
```

> This will make all variables and mixins available in the file you are working on. The `as *` is important as it makes everything in the library available without a namespace.

### Requirements

We are using the latest version of Dart Sass. [Please review all documentation(https://sass-lang.com/documentation)] as there are some minor differences with LibSass and Ruby Sass which are older.

## What's Available

### Variables

All variables that are created from `@espressive/design-tokens` are available once declared with `@use`.

### Mixins & Helpers

There are a number of helpers available with scopes.

#### Media Queries

```
.ResponsiveModule:before {
  content: "tiny";

  @include media-xs {
    content: "small";
  }

  @include media-md {
    content: "medium";
  }

  @include media-lg {
    content: "large";
  }

  @include media-xl {
    content: "extra large";
  }
}
```

#### Scroll

```
.ScrollingModule {
  @include scroll-all;
  // or
  @include scroll-horizontal;
  // or
  @include scroll-vertical;
}
```
