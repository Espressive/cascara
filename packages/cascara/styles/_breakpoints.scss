// @use '@espressive/design-tokens/scss' as *;
// Bug in design tokens causing PX vals to get stripped and replaced with rem without conversion :/

$size-screen-0: 320px;
$size-screen-1: 768px;
$size-screen-2: 992px;
$size-screen-3: 1400px;
$size-screen-4: 1900px;

// These functions should not be needed once our library compiles correctly
@function em($number) {
  @return ($number/16)+0em;
}

@function emFromPx($number) {
  @if type-of($number) == 'number' and not unitless($number) {
    @return em($number / ($number * 0 + 1));
  }

  @return em($number);
}

// Mobile first queries (greater than)
@mixin gt-xs {
  @media screen and (min-width: #{emFromPx($size-screen-0 + 1px)}) {
    @content;
  }
}

@mixin gt-sm {
  @media screen and (min-width: #{emFromPx($size-screen-1 + 1px)}) {
    @content;
  }
}

@mixin gt-md {
  @media screen and (min-width: #{emFromPx($size-screen-2 + 1px)}) {
    @content;
  }
}

@mixin gt-lg {
  @media screen and (min-width: #{emFromPx($size-screen-3 + 1px)}) {
    @content;
  }
}

@mixin gt-xl {
  @media screen and (min-width: #{emFromPx($size-screen-4 + 1px)}) {
    @content;
  }
}

// Queries for only one breakpoint
@mixin xs {
  @media screen and (max-width: #{emFromPx($size-screen-0)}) {
    @content;
  }
}

@mixin sm {
  @media screen and (min-width: #{emFromPx($size-screen-0 + 1px)}) and (max-width: #{emFromPx($size-screen-1)}) {
    @content;
  }
}

@mixin md {
  @media screen and (min-width: #{emFromPx($size-screen-1 + 1px)}) and (max-width: #{emFromPx($size-screen-2)}) {
    @content;
  }
}

@mixin lg {
  @media screen and (min-width: #{emFromPx($size-screen-2 + 1px)} ) and (max-width: #{emFromPx($size-screen-3)}) {
    @content;
  }
}

@mixin xl {
  @media screen and (min-width: #{emFromPx($size-screen-3 + 1px)}) and (max-width: #{emFromPx($size-screen-4)}) {
    @content;
  }
}
