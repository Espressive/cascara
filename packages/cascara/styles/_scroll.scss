@use '~@espressive/design-tokens/scss' as *;

%scrollbars {
  -webkit-overflow-scrolling: touch;
  scroll-behavior: smooth;

  &::-webkit-scrollbar {
    width: 0;
  }

  &::-webkit-scrollbar-thumb {
    background-clip: content-box;
    background-color: $color-base-gray-medium;
    border: 0.25rem solid rgba(0, 0, 0, 0);
    border-radius: 1rem;
    box-shadow: inset 0 0 0 0.05em rgba(255, 255, 255, 0.2);
  }

  &:hover {
    &::-webkit-scrollbar {
      width: 0.875rem;
    }
  }
}

@mixin vertical {
  overflow-y: overlay;

  @extend %scrollbars;
}

@mixin horizontal {
  overflow-x: overlay;

  @extend %scrollbars;
}

@mixin all {
  overflow: overlay;

  @extend %scrollbars;
}
