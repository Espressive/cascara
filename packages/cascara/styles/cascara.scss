@forward '@espressive/design-tokens/scss';
@forward 'breakpoints' as media-*;
@forward 'scroll' as scroll-*;
