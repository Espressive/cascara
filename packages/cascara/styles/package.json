{
  "name": "@espressive/cascara/styles",
  "private": true,
  "sideEffects": false,
  "main": "cascara.scss",
  "css": "cascara.scss"
}
