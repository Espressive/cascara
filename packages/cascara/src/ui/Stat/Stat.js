import React, { useCallback } from 'react';
import pt from 'prop-types';
import classNames from 'classnames/bind';

import { Clickable, Role } from 'reakit';
import styles from './Stat.module.scss';

const propTypes = {
  /** Stats can have their own css class name */
  className: pt.string,
  /** Stats can be fuild */
  fluid: pt.bool,
  /** Stats can have a label */
  label: pt.string,
  /** Stats can be clickable */
  onClick: pt.func,
  /** Stats can have a sub text */
  sub: pt.oneOfType([pt.number, pt.string]),
  /** Stats can have a value */
  value: pt.oneOfType([pt.number, pt.string]),
};

const cx = classNames.bind(styles);

const WidgetStatsStat = ({ className, fluid, onClick, label, value, sub }) => {
  // Instead of maintaining separate, competing styles for focus, we are setting focus on this clickable item on hover. This may be something we consider doing on other Clickable components with Reakit.
  const handleFocus = useCallback(({ currentTarget, dispatchConfig }) => {
    const { registrationName } = dispatchConfig;

    if (registrationName === 'onMouseEnter') {
      currentTarget.focus();
    } else if (registrationName === 'onMouseLeave') {
      currentTarget.blur();
    }
  }, []);

  return (
    <Role
      as={onClick ? Clickable : 'div'}
      className={cx('Stat', className, {
        Clickable: Boolean(onClick),
        Fluid: fluid,
      })}
      disabled={!onClick}
      focusable={Boolean(onClick)}
      key={label}
      onClick={onClick}
      onMouseEnter={handleFocus}
      onMouseLeave={handleFocus}
    >
      <span className={styles.Value}>{value}</span>
      <h4 className={styles.Label}>{label}</h4>
      {sub && <span className={styles.Sub}>{sub}</span>}
    </Role>
  );
};

WidgetStatsStat.propTypes = propTypes;

export { propTypes };
export default WidgetStatsStat;
