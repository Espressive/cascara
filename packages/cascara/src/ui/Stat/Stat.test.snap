// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stat snapshot tests 1`] = `
<div>
  <div
    class="Stat"
    data-component="Stat"
    disabled=""
    focusable="false"
  >
    <span
      class="Value"
    >
      50
    </span>
    <h4
      class="Label"
    >
      Stat
    </h4>
  </div>
</div>
`;

exports[`stat snapshot tests 2`] = `
<div>
  <button
    class="Stat Clickable"
    data-component="Stat"
  >
    <span
      class="Value"
    >
      50
    </span>
    <h4
      class="Label"
    >
      Stat
    </h4>
  </button>
</div>
`;

exports[`stat snapshot tests 3`] = `
<div>
  <div
    class="Stat Fluid"
    data-component="Stat"
    disabled=""
    focusable="false"
  >
    <span
      class="Value"
    >
      50
    </span>
    <h4
      class="Label"
    >
      Stat
    </h4>
  </div>
</div>
`;

exports[`stat snapshot tests 4`] = `
<div>
  <div
    class="Stat"
    data-component="Stat"
    disabled=""
    focusable="false"
  >
    <span
      class="Value"
    >
      50
    </span>
    <h4
      class="Label"
    >
      Stat
    </h4>
    <span
      class="Sub"
    >
      Component
    </span>
  </div>
</div>
`;
